byoeb-v1/byoeb/byoeb/services/auth/oauth_provider.py

```
@@ -0,0 +1,453 @@
+            Route("/.well-known/oauth-authorization-server", endpoint=cors_middleware(MetadataHandler(metadata).handle, ["GET", "OPTIONS"]), methods=["GET", "OPTIONS"]),
+            Route("/oauth/authorize", endpoint=self._authorize, methods=["GET", "POST"]),
+            Route("/oauth/token", endpoint=cors_middleware(self._token, ["POST", "OPTIONS"]), methods=["POST", "OPTIONS"]),
+        ]
+        if self._client_registration_options.enabled:
+            registration_handler = RegistrationHandler(self, options=self._client_registration_options)
+            routes.append(Route("/oauth/register", endpoint=cors_middleware(registration_handler.handle, ["POST", "OPTIONS"]), methods=["POST", "OPTIONS"]))
+        return routes
+
+    def create_resource_routes(self, resource_path: str, scopes: list[AuthPermission]) -> list[Route]:
+        resource_url = self._get_resource_url(resource_path)
+        if not resource_url: raise RuntimeError("Cannot create resource URL")
```

Ruff (0.15.11)

[error] Multiple statements on one line (colon)

(E701)